### PR TITLE
update Select SelectMulti and Combobox replace readOnly with inputReadOnly

### PR DESCRIPTION
**Pull request by laraharrow**
### :sparkles: Changes

- update Select SelectMulti and Combobox replace readOnly with inputReadOnly
- InputSearch now supports disabled and readOnly - updated docs and gatsby
- InputChips now supports disabled and readOnly - updated docs and gatsby

### :white_check_mark: Requirements

- [x] Includes test coverage for all changes
- [x] Build and tests are passing
- [x] Update documentation
- [x] Updated CHANGELOG
- [ ] Checked for i18n impacts
- [ ] Checked for a11y impacts
- [x] Check for image-snapshot changes (run `yarn image-snapshots` locally)
- [x] PR is ideally < ~400LOC

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ NOTE: This is the CHANGELOG for the @looker/components MONOREPO. Each package ha
 
 ### Fixed
 
+- update Select SelectMulti and Combobox replace readOnly with inputReadOnly
```
Comment (mdodgelooker) [inline CHANGELOG.md:21]: No need to update the root `CHANGELOG.md`, just `components/CHANGELOG.md`
```suggestion
```

File: packages/components/src/Form/Inputs/InputChips/InputChipsBase.tsx [hunk 2]
```diff
@@ -126,6 +128,7 @@ export const InputChipsBaseInternal = forwardRef(
       isVisibleOptions,
       showCaret = false,
       isClearable = true,
+      readOnly,
```
Comment (mdodgelooker) [inline packages/components/src/Form/Inputs/InputChips/InputChipsBase.tsx:131]: One thing I forgot about is you can delete chips via `Backspace` key when the input is focused, or when "select" a chip or chips. Adding `&& !readOnly` below in ln 251 and `if (!readOnly) {` after ln 181 should do the trick.

**Review by mdodgelooker — CHANGES_REQUESTED** [review]
Just realized the Backspace key still deletes chips with `readOnly` on, added a suggestion for how to disable that.

File: packages/components/src/Form/Inputs/InputChips/InputChipsBase.tsx (hunk 2 above)
Comment (laraharrow) [reply packages/components/src/Form/Inputs/InputChips/InputChipsBase.tsx:131]: @mdodgelooker I don't think I understand the changes. I don't see anything to be changed on line 181. and on 131 I don't think I can add the syntax `if (!readOnly)`.

Comment (mdodgelooker) [reply packages/components/src/Form/Inputs/InputChips/InputChipsBase.tsx:131]: Hi @laraharrow I created a mini-PR on top of this branch here https://github.com/looker-open-source/components/pull/1657 to disable deleting of chips via keyboard when `readOnly` is true.

Comment (laraharrow) [reply packages/components/src/Form/Inputs/InputChips/InputChipsBase.tsx:131]: Great, oh I see yeah now it all make sense! sorry for the confusion!
Thanks

File: www/src/documentation/components/forms/input-chips.mdx [hunk 3]
```diff
@@ -165,3 +165,28 @@ When set to false, hitting the `Backspace` key while focused in the input will n
...
+
+Use the `disable` property to make an input field un-editable.
+
+```jsx
+<InputChips disabled values={['disbled']} />
+<InputChips
+  placeholder="this is a disabled placeholder"
+  disabled
+  values={[]}
+/>
```
Comment (mdodgelooker) [inline www/src/documentation/components/forms/input-chips.mdx:179]:
```suggestion
<SpaceVertical>
  <InputChips disabled values={['disabled']} />
  <InputChips
    placeholder="this is a disabled placeholder"
    disabled
    values={[]}
  />
</SpaceVertical>
```

**Review by mdodgelooker — CHANGES_REQUESTED** [review]
I have a CSS-related suggestion and a docs-related one, and then I made another mini-PR here https://github.com/looker-open-source/components/pull/1661 for the changes needed in the `./Combobox` folder to use `inputReadOnly` for a couple behaviors that were relying on `readOnly` before.

**Comment by mdodgelooker** [comment]
The reason all the snapshots are changed is I think due to the styled-components 5 upgrade. If you run `yarn` and then update the image snapshots again, I think they'll mostly revert except for the ones that are affected by this PR.

Also, I just merged my Focus Trap PR and gave you some more conflicts. Sorry!
FYI, in `useInputEvents.ts`, the bit of code that is updated in this PR has now been completely removed, so after rebasing, that file shouldn't show any changes in this PR.

File: CHANGELOG.md [hunk 4]
```diff
@@ -14,6 +14,14 @@ NOTE: This is the CHANGELOG for the @looker/components MONOREPO. Each package ha
...
 - [Icons](./packages/icons/CHANGELOG.md)
 
+## [UNRELEASED]
+
+### Fixed
+
+update Select SelectMulti and Combobox replace readOnly with inputReadOnly
+InputSearch now supports disabled and readOnly - updated docs and gatsby
+InputChips now supports disabled and readOnly - updated docs and gatsby
+
```
Comment (mdodgelooker) [inline CHANGELOG.md:24]: No need to update this file, just the one in `./packages/components`
```suggestion
```

File: packages/components/src/Form/Inputs/InputChips/InputChipsBase.tsx [hunk 5]
```diff
@@ -365,7 +373,7 @@ export const InputChipsBaseInternal = forwardRef(
         onFocus={wrappedOnFocus}
         onKeyDown={wrappedOnKeyDown}
         validationType={validationType}
-        height="auto"
+        readOnly={readOnly || inputReadOnly}
```
Comment (mdodgelooker) [inline packages/components/src/Form/Inputs/InputChips/InputChipsBase.tsx:376]:
```suggestion
        readOnly={readOnly || inputReadOnly}
        height="auto"
```

File: www/src/documentation/components/forms/input-chips.mdx [hunk 6]
```diff
@@ -165,3 +165,30 @@ When set to false, hitting the `Backspace` key while focused in the input will n
...
 }
 ```
+
+## Disabled
+
+Use the `disable` property to make an input field un-editable.
+
+```jsx
+<SpaceVertical>
+  <InputChips disabled values={['disbled']} />
```
Comment (mdodgelooker) [inline www/src/documentation/components/forms/input-chips.mdx:175]:
```suggestion
  <InputChips disabled values={['disabled']} />
```

**Review by mdodgelooker — CHANGES_REQUESTED** [review]
Hey there, found a few things, perhaps due to the merge.

**Review by mdodgelooker — APPROVED** [review]
I forgot to mention that a couple snapshots were affected by the `height="auto"` prop in `InputChipsBase` – I just updated and pushed them. This is awesome, great work!

**Pull request merged**